### PR TITLE
More e2e test fixes

### PR DESCRIPTION
**Pull request by donatwork**
# Description
Updated e2e tests in areas involving PowerScale and PowerFlex for Observability with Authorization. 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/853|

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Ran modified tests to make sure they can pass.

### CODE REVIEW

**Review by bharathsreekanth — APPROVED** [review]

**Review by gallacher — APPROVED** [review]
I approve these test changes

File: tests/e2e/testfiles/values.yaml [hunk 1]
```diff
@@ -1,4 +1,19 @@
 # List of E2E Tests Scenarios
+- scenario: "Install Authorization Proxy Server"
+  paths:
+    - "testfiles/authorization-templates/csm_authorization_proxy_server.yaml"
+  modules:
+    - "authorizationproxyserver"
+    - "authorization"
```
Comment (shaynafinocchiaro) [inline tests/e2e/testfiles/values.yaml:7]: This is the sidecar value. Since no driver is being installed, it's not required here

**Comment by bharathsreekanth** [comment]
All changes in the PR seem to be related to test fixes, should not affect Operator functionality. LGTM

File: tests/e2e/testfiles/values.yaml [hunk 2]
```diff
@@ -361,25 +380,35 @@
...
+    - "Validate custom resource [2]"
+    - "Validate [powerscale] driver from CR [2] is installed"
+    - "Validate [authorization] module from CR [2] is installed"
+    - "Validate [observability] module from CR [2] is installed"
     - "Run custom test"
     # cleanup
-    - "Enable forceRemoveDriver on CR [1]"
-    - "Delete custom resource [1]"
+    - "Enable forceRemoveDriver on CR [2]"
+    - "Delete custom resource [2]"
```
Comment (shaynafinocchiaro) [inline tests/e2e/testfiles/values.yaml:407]: Missing clean up steps to delete authorization server and restoring templates

File: tests/e2e/testfiles/values.yaml [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
+    - "authorization"
+  steps:
+    - "Given an environment with k8s or openshift, and CSM operator installed"
+    - "Create [authorization-proxy-server] prerequisites from CR [1]"
+    - "Apply custom resource [1]"
+    - "Validate [authorization-proxy-server] module from CR [1] is installed"
+    - "Configure authorization-proxy-server for [powerscale]"
+    - "Configure authorization-proxy-server for [powerflex]"
+    - "Delete custom resource [1]"
+
```
Comment (donatwork) [inline tests/e2e/testfiles/values.yaml:16]: This test was initially at the bottom. Felt it was important enough to get it run before the rest of the scenarios which depend on it.

File: tests/e2e/testfiles/values.yaml [hunk 4]
```diff
@@ -420,57 +449,87 @@
...
+    - "Set secret for driver from CR [2] to [isilon-creds]"
+    - "Validate [powerscale] driver from CR [2] is installed"
+    - "Validate [authorization] module from CR [2] is not installed"
+    - "Validate [observability] module from CR [2] is not installed"
+    - "Run custom test"
     # cleanup
-    - "Enable forceRemoveDriver on CR [1]"
-    - "Delete custom resource [1]"
+    - "Enable forceRemoveDriver on CR [2]"
+    - "Delete custom resource [2]"
```
Comment (shaynafinocchiaro) [inline tests/e2e/testfiles/values.yaml:487]: Missing clean up steps to delete authorization server and restoring templates

File: tests/e2e/testfiles/values.yaml [hunk 5]
```diff
@@ -420,57 +449,87 @@
...
+    - "Disable [observability] module from CR [2]"
+    - "Validate [powerscale] driver from CR [2] is installed"
+    - "Validate [authorization] module from CR [2] is not installed"
+    - "Validate [observability] module from CR [2] is not installed"
+    - "Run custom test"
     # cleanup
-    - "Enable forceRemoveDriver on CR [1]"
-    - "Delete custom resource [1]"
+    - "Enable forceRemoveDriver on CR [2]"
+    - "Delete custom resource [2]"
```
Comment (shaynafinocchiaro) [inline tests/e2e/testfiles/values.yaml:528]: Missing clean up steps to delete authorization server and restoring templates

**Review by sharmilarama — APPROVED** [review]

**Review by rensyct — APPROVED** [review]

File: tests/e2e/testfiles/values.yaml (hunk 1 above)
Comment (donatwork) [reply tests/e2e/testfiles/values.yaml]: Removed sidecar.

File: tests/e2e/testfiles/values.yaml (hunk 2 above)
Comment (donatwork) [reply tests/e2e/testfiles/values.yaml:407]: Added cleanup.

File: tests/e2e/testfiles/values.yaml (hunk 4 above)
Comment (donatwork) [reply tests/e2e/testfiles/values.yaml:494]: Added cleanup.

File: tests/e2e/testfiles/values.yaml (hunk 5 above)
Comment (donatwork) [reply tests/e2e/testfiles/values.yaml:541]: Added cleanup.

**Review by shaynafinocchiaro — APPROVED** [review]

**Review by harshaatdell — APPROVED** [review]

**Pull request merged**